cmd/honeyalb/main.go

```diff
@@ -0,0 +1,230 @@
+			downloadsCh := make(chan state.DownloadedObject)
+
+			// For now, just run one goroutine per-LB
+			for _, lbName := range lbNames {
+				logrus.WithFields(logrus.Fields{
+					"lbName": lbName,
+				}).Info("Attempting to ingest ALB")
+
+				elbSvc := elbv2.New(sess, nil)
+
+				lbNameResp, err := elbSvc.DescribeLoadBalancers(&elbv2.DescribeLoadBalancersInput{
+					Names: []*string{
+						aws.String(lbName),
+					},
+				})
+				if err != nil {
+					fmt.Fprintln(os.Stderr, err)
+					os.Exit(1)
+				}
+
+				lbArn := lbNameResp.LoadBalancers[0].LoadBalancerArn
+				lbArnResp, err := elbSvc.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+					LoadBalancerArn: lbArn,
+				})
+				if err != nil {
+					fmt.Fprintln(os.Stderr, err)
+					os.Exit(1)
+				}
+
+				enabled := false
+				bucketName := ""
+				bucketPrefix := ""
+
+				for _, element := range lbArnResp.Attributes {
+					if *element.Key == "access_logs.s3.enabled" && *element.Value == "true" {
+						enabled = true
+					}
+					if *element.Key == "access_logs.s3.bucket" {
+						bucketName = *element.Value
+					}
+					if *element.Key == "access_logs.s3.prefix" {
+						bucketPrefix = *element.Value
+					}
+				}
+
+				if !enabled {
+					fmt.Fprintf(os.Stderr, `Access logs are not configured for ALB %q. Please enable them to use the ingest tool.
+
+For reference see this link:
+
+http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging
+`, lbName)
+					os.Exit(1)
+				}
+				logrus.WithFields(logrus.Fields{
+					"bucket": bucketName,
+					"lbName": lbName,
+				}).Info("Access logs are enabled for ALB ♥")
+
+				albDownloader := logbucket.NewALBDownloader(sess, bucketName, bucketPrefix, lbName)
+				downloader := logbucket.NewDownloader(sess, stater, albDownloader, opt.BackfillHr)
+
+				// TODO: One-goroutine-per-LB feels a bit
+				// silly.
+				go downloader.Download(downloadsCh)
+			}
+
+			signalCh := make(chan os.Signal)
+			signal.Notify(signalCh, os.Interrupt)
+
+			go func() {
+				<-signalCh
+				logrus.Fatal("Exiting due to interrupt.")
+			}()
+
+			for {
+				download := <-downloadsCh
+				if err := defaultPublisher.Publish(download); err != nil {
+					logrus.WithField("object", download).Error("Cannot properly publish downloaded object")
+				}
+			}
+		}
+	}
+
+	return fmt.Errorf("Subcommand %q not recognized", args[0])
+}
+
+func main() {
+	flagParser := flag.NewParser(opt, flag.Default)
+	args, err := flagParser.Parse()
+	if err != nil {
+		os.Exit(1)
+	}
+
+	if opt.Debug {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+
+	formatter := &logrus.TextFormatter{
+		FullTimestamp: true,
+	}
+	logrus.SetFormatter(formatter)
+
+	logrus.WithField("version", BuildID).Debug("Program starting")
+
+	if opt.Dataset == "aws-$SERVICE-access" {
+		opt.Dataset = "aws-elb-access"
```

Comment:
alb

Reply:
We had decided sending ALB and ELB to the same place is OK [here](https://github.com/honeycombio/honeyaws/pull/33#discussion_r169720980) - do you want to stick with that (same dataset)? I don't really have a preference, though I do find not having a `aws-alb-access` a bit confusing as an end user.